pkg/runtime/processor/pubsub/subscribe.go

```diff
@@ -133,35 +148,72 @@ func (p *pubsub) subscriptions(ctx context.Context) ([]rtpubsub.Subscription, er
 	}
 
 	appChannel := p.channels.AppChannel()
-	if appChannel == nil {
-		log.Warn("app channel not initialized, make sure -app-port is specified if pubsub subscription is required")
-		return nil, nil
-	}
 
 	var (
 		subscriptions []rtpubsub.Subscription
 		err           error
 	)
 
-	// handle app subscriptions
-	if p.isHTTP {
-		subscriptions, err = rtpubsub.GetSubscriptionsHTTP(ctx, appChannel, log, p.resiliency)
-	} else {
-		var conn grpc.ClientConnInterface
-		conn, err = p.grpc.GetAppClient()
+	// handle app subscriptions, if app channel is initialised
+	if appChannel != nil {
+		if p.isHTTP {
+			subscriptions, err = rtpubsub.GetSubscriptionsHTTP(ctx, appChannel, log, p.resiliency)
+		} else {
+			var conn grpc.ClientConnInterface
+			conn, err = p.grpc.GetAppClient()
+			if err != nil {
+				return nil, fmt.Errorf("error while getting app client: %w", err)
+			}
+			client := runtimev1pb.NewAppCallbackClient(conn)
+			subscriptions, err = rtpubsub.GetSubscriptionsGRPC(ctx, client, log, p.resiliency)
+		}
 		if err != nil {
-			return nil, fmt.Errorf("error while getting app client: %w", err)
+			return nil, err
 		}
-		client := runtimev1pb.NewAppCallbackClient(conn)
-		subscriptions, err = rtpubsub.GetSubscriptionsGRPC(ctx, client, log, p.resiliency)
 	}
-	if err != nil {
-		return nil, err
+
+	// inject auto-generated canary subscriber
+	cs := p.autoGenerateCanarySubscriber
+	if cs != "" {
+		parts := strings.Split(cs, ":")
+		if len(parts) != 2 {
+			// TODO: Move this logic as early as possible in the init phase
+			panic("'autoGenerateCanarySubscriber' Should have had two parts")
```

Comment:
Should this panic here or could we potentially handle this gracefully with an error?

Reply:
If this code path is reached, the user has misconfigured dapr, so we must instruct dapr to exit (this is why I used panic) - I'm very new to Go, so if there is a better way to do that please do let me know :)